src/nlp_expr.jl

```diff
@@ -1111,3 +1111,34 @@ end
 function MOI.VectorNonlinearFunction(f::Vector{<:AbstractJuMPScalar})
     return MOI.VectorNonlinearFunction(map(moi_function, f))
 end
+
+# LinearAlgebra overloads to throw nicer error messages. These may be changed to
+# return expressions in the future.
+
+function LinearAlgebra.det(::AbstractMatrix{<:AbstractJuMPScalar})
+    return throw(MOI.UnsupportedNonlinearOperator(:det))
+end
+
+function LinearAlgebra.logdet(::AbstractVector{<:AbstractJuMPScalar})
```

Comment:
`logdet` needs a square matrix so I don't think we need this one

Reply:
`logdet` has a generic fallback to `det`. I guess we need a rule about whether to intercept it or not
```julia
julia> using LinearAlgebra

julia> x = rand(3)
3-element Vector{Float64}:
 0.7765560317032483
 0.9331828725371808
 0.2083409558020034

julia> logdet(x)
ERROR: MethodError: no method matching det(::Vector{Float64})
```

Reply:
What I mean is that we don't need to intercept any of `logdet` or `det` if the input is a vector